### ISSUE TITLE
REST Client shoud implement an client interface

### ISSUE THREAD

**Issue by hsanjuan** [ipfs/ipfs-cluster#447]
#### Basic information

* [x] Type (mark as appropiate):
  * [ ] Bug
  * [x] Feature request
  * [ ] Enhancement

#### Description

Our `api/rest/` module should provide an interface to be implemented by API Clients (perhaps as separate `api/rest/client_interface` module. The `api/rest/client` client should be an implementation of that interface.

**Comment by AT1452** [comment]
How can I help? Any specifications I should follow?

**Comment by hsanjuan** [comment]
The spec is the current `api/rest/client` module. It should implement an interface (which matches the methods it exposes). We just need to write that interface (which is just made from the public methods of the client).

**Comment by AT1452** [comment]
I'm thinking of starting with a simple HTTP wireframe, is that alright?

**Comment by hsanjuan** [comment]
@AT1452 I think you are misunderstanding.

Currently there is a `Client`, which just has some methods but is not implementing any interface. I just want to have an interface which is fulfilled by this `Client`. For example, the `Client` has:

`func (c *Client) ID() (api.ID, error) { .. }`

So we want an interface:

```
type ClientIface interface {
...
ID() (api.ID, error)
...
}

This issue is about just writing that interface (and then using it in a couple of places, instead of the concrete `Client` implementation).

**Comment by hsanjuan** [comment]
Fixed by #523
